### ISSUE TITLE
Seurat v3.0 DimPlot ncol specification issue

### ISSUE THREAD

**Issue by mosquitoCat** [satijalab/seurat#1535]
Hi there,

I was trying to use DimPlot with split.by and ncol specification to show two groups of 4 (8 in total) datasets and found that for some reasons the ncol spec was not picked up. At the beginning, I thought it was because of the last if statement in the CombinePlots function [https://github.com/satijalab/seurat/blob/25b830b0dd6f12538516f0faf9a3b3ddfd0ce6d8/R/visualization.R](url) line 674 - 676. I swapped "ncol" and "1" but it still didn't pick up the spec. Then I added the nrow spec from facet_wrap to the main function and it worked this time. It didn't solved the ncol issue but is a temporary solution. 

Could you maybe have a look and fix it please?

Thank you

**Comment by andrewwbutler** [comment]
Hi,

Currently if you use `split.by`, each split will get it's own column (and the `ncol` parameter will be ignored). This is useful when combined with multiple `group.by` categories to compare the splits across different groupings. Can you provide an example plot + code of the current behavior and the expected (or requested) behavior?

**Comment by mosquitoCat** [comment]
Hi Andrew,

Thank you for getting back to me. I use both split.by for the 8 single datasets and the group.by for the 2 treatment group I have. I'd like to have 8 single plots with 4 datasets having the same condition to be in the same row (2x4). 

The code I tried to achieve this is:

`DimPlot(toy, reduction = "umap", label = F, do.returen = T, combine = T, split.by = "dataset", group.by = "condition", ncol = 4)`

![Fig1](https://user-images.githubusercontent.com/7779719/57940532-8f067900-78c4-11e9-8a24-b7262d64a5b8.png)

So I modified the function to allow me to pass the row number specs and got the graph below, which is what I wanted. The code is:

`DimPlot_Cass(toy, reduction = "umap", label = F, do.returen = T, combine = T, split.by = "dataset", group.by = "condition", nrow = 2)`

![Fig2](https://user-images.githubusercontent.com/7779719/57940629-c83ee900-78c4-11e9-84a6-a069d022404b.png)

Looking forward to your further help.

Thank you,
Cass
